ui/src/app/views/account-creation/component.spec.ts

```diff
@@ -69,60 +50,92 @@ describe('AccountCreationComponent', () => {
           })
         }]
     }).compileComponents().then(() => {
-      page = new AccountCreationPage(TestBed);
+      fixture = TestBed.createComponent(AccountCreationComponent);
       tick();
+      updateAndTick(fixture);
     });
   }));
 
   it('handles selecting username', fakeAsync(() => {
-    page.readPageData();
-    expect(page.component.usernameFocused).toBeFalsy();
-    simulateEvent(page.fixture, page.usernameField, 'focus');
-    expect(page.component.usernameFocused).toBeTruthy();
-    simulateEvent(page.fixture, page.usernameField, 'blur');
-    expect(page.component.usernameFocused).toBeFalsy();
+    expect(fixture.componentInstance.usernameFocused).toBeFalsy();
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#username')), 'focus');
+    expect(fixture.componentInstance.usernameFocused).toBeTruthy();
+    simulateEvent(fixture, fixture.debugElement.query(By.css('#username')), 'blur');
+    expect(fixture.componentInstance.usernameFocused).toBeFalsy();
   }));
 
   it('handles each username invalidity', fakeAsync(() => {
-    page.readPageData();
     // Begin with a period
-    simulateEvent(page.fixture, page.usernameField, 'focus');
-    simulateInput(page.fixture, page.usernameField, '.username');
+    const usernameField = fixture.debugElement.query(By.css('#username'));
+    simulateEvent(fixture, usernameField, 'focus');
+    simulateInput(fixture, usernameField, '.username');
     tick(300);
-    updateAndTick(page.fixture);
-    expect(page.fixture.debugElement.query(By.css('#username-invalid-error'))).toBeTruthy();
-    expect(page.usernameField.classes.unsuccessfulInput).toBeTruthy();
+    updateAndTick(fixture);
+    expect(fixture.debugElement.query(By.css('#username-invalid-error'))).toBeTruthy();
```

Comment:
This is one of the 'this could crash jasmine' query expects. Since this was just refactoring out the page object, probably okay to leave, but just wanted to show you what it looked like. And also it may make sense to throw a tech debt ticket for this in the backlog as well.

Reply:
Ah ... I think this was one of the ones I refactored ... will update.

Reply:
Done.